crates/router/src/types/api.rs

```diff
@@ -255,6 +255,9 @@ impl ConnectorData {
                 enums::Connector::Trustpay => Ok(Box::new(&connector::Trustpay)),
                 enums::Connector::Tsys => Ok(Box::new(&connector::Tsys)),
                 enums::Connector::Zen => Ok(Box::new(&connector::Zen)),
+                _ => Err(report!(errors::ConnectorError::InvalidConnectorName)
```

Comment:
Avoid using underscore to handle invalid cases. Specific the connector name and handle the cases.